usecases/metrics_ingestion.go

```
@@ -0,0 +1,72 @@
+}
+
+type MetricsIngestionUsecase struct {
+	metricRepository  MetricsIngestionRepository
+	licenseRepository publicLicenseRepository
+	executorFactory   executor_factory.ExecutorFactory
+}
+
+func NewMetricsIngestionUsecase(
+	metricRepository MetricsIngestionRepository,
+	licenseRepository publicLicenseRepository,
+	executorFactory executor_factory.ExecutorFactory,
+) MetricsIngestionUsecase {
+	return MetricsIngestionUsecase{
+		metricRepository:  metricRepository,
+		licenseRepository: licenseRepository,
+		executorFactory:   executorFactory,
+	}
+}
+
+func (u *MetricsIngestionUsecase) IngestMetrics(ctx context.Context, collection models.MetricsCollection) error {
+	logger := utils.LoggerFromContext(ctx)
+
+	if collection.LicenseKey != nil {
+		logger.DebugContext(ctx, "Checking license")
+		license, err := u.validateLicense(ctx, *collection.LicenseKey)
+		if err != nil {
+			logger.ErrorContext(ctx, "Failed to validate license", "error", err.Error())
+			return errors.Wrap(models.UnAuthorizedError, "invalid license")
+		}
+		collection.LicenseName = &license.OrganizationName
+	}
+
+	err := u.metricRepository.SendMetrics(ctx, collection)
+	if err != nil {
+		logger.ErrorContext(ctx, "Failed to send metrics to BigQuery", "error", err.Error())
+		return fmt.Errorf("failed to send metrics to BigQuery: %w", err)
+	}
+
+	return nil
+}
+
+// Only check if the license exists
+func (u *MetricsIngestionUsecase) validateLicense(ctx context.Context, licenseKey string) (models.License, error) {
+	logger := utils.LoggerFromContext(ctx)
+
+	license, err := u.licenseRepository.GetLicenseByKey(ctx,
+		u.executorFactory.NewExecutor(), licenseKey)
+	if err != nil {
+		if !errors.Is(err, models.NotFoundError) {
+			utils.LogAndReportSentryError(ctx, err)
+		}
+
+		logger.WarnContext(ctx, "Invalid license", "license", licenseKey)
+		return models.License{}, errors.Wrap(models.UnAuthorizedError, "invalid license")
+	}
+	return license, nil
+}
```

@apognu @Pascal-Delange Should we raise an error and return a 401 error ? 

If not, what is the right behavior? Ignore and send the bad license key in BQ ?

---

I think I had a chat about this but it must have been with Antoine ?
Anyway, I think we maybe log an info log, but mostly just ignore the invalid license key and treat it as a call without license key.